webapp/packages/plugin-codemirror6/src/ReactCodemirror.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { EditorView } from 'codemirror6';
+import { forwardRef, useEffect, useImperativeHandle, useLayoutEffect, useState } from 'react';
+
+import { EditorState, Annotation, StateEffect } from '@codemirror/state';
+import type { ViewUpdate } from '@codemirror/view';
+
+import { getDefaultExtensions } from './getDefaultExtensions';
+import type { IEditorRef } from './IEditorRef';
+import type { IReactCodeMirrorProps } from './IReactCodemirrorProps';
+
+const External = Annotation.define<boolean>();
+
+export const ReactCodemirror = forwardRef<IEditorRef, IReactCodeMirrorProps>(function ReactCodemirror({
+  value,
+  extensions = [],
+  readonly,
+  editable,
+  autoFocus,
+  onChange,
+  onUpdate,
+}, ref) {
+  const [container, setContainer] = useState<HTMLDivElement | null>(null);
+  const [view, setView] = useState<EditorView | null>(null);
+  const [state, setState] = useState<EditorState | null>(null);
+
+  const defaultExtensions = getDefaultExtensions({
+    readonly,
+    editable,
+  });
```

Comment:
probably we want to have them as property, in case when i want to define custom extensions or configure existing